tidy3d/plugins/autograd/invdes/penalties.py

```diff
@@ -1,86 +1,140 @@
 
     Parameters
     ----------
-    filter_size : Tuple[int, ...]
-        The size of the filter to be used for erosion and dilation.
-    beta : float, optional
-        Strength of the tanh projection. Default is 100.0.
-    eta : float, optional
-        Midpoint of the tanh projection. Default is 0.5.
-    delta_eta : float, optional
-        The binarization threshold for erosion and dilation operations. Default is 0.01.
-    padding : PaddingType, optional
-        The padding type to use for the filter. Default is "reflect".
-
-    Returns
-    -------
-    Callable
-        A function that computes the erosion/dilation penalty for a given array.
+    filtproj : FilterAndProject
+        The filter and project instance to use for erosion and dilation operations.
+    delta_eta : float = 0.01
+        The binarization threshold for erosion and dilation operations.
     """
-    filtproj = make_filter_and_project(filter_size, beta, eta, padding=padding)
-    eta_dilate = 0.0 + delta_eta
-    eta_eroded = 1.0 - delta_eta
 
-    def _dilate(array: np.ndarray, beta: float):
-        return filtproj(array, beta=beta, eta=eta_dilate)
+    filtproj: FilterAndProject
+    delta_eta: float = 0.01
+
+    def __call__(self, array: NDArray) -> float:
+        """Compute the erosion/dilation penalty for a given array.
+
+        Parameters
+        ----------
+        array : NDArray
+            The input array to compute the penalty for.
 
-    def _erode(array: np.ndarray, beta: float):
-        return filtproj(array, beta=beta, eta=eta_eroded)
+        Returns
+        -------
+        float
+            The computed erosion/dilation penalty.
+        """
+        eta_dilate = 0.0 + self.delta_eta
+        eta_eroded = 1.0 - self.delta_eta
 
-    def _open(array: np.ndarray, beta: float):
-        return _dilate(_erode(array, beta=beta), beta=beta)
+        def _dilate(arr: NDArray):
+            return self.filtproj(arr, eta=eta_dilate)
 
-    def _close(array: np.ndarray, beta: float):
-        return _erode(_dilate(array, beta=beta), beta=beta)
+        def _erode(arr: NDArray):
+            return self.filtproj(arr, eta=eta_eroded)
 
-    def _erosion_dilation_penalty(array: np.ndarray, beta: float = beta):
-        diff = _close(array, beta) - _open(array, beta)
+        def _open(arr: NDArray):
+            return _dilate(_erode(arr))
+
+        def _close(arr: NDArray):
+            return _erode(_dilate(arr))
+
+        diff = _close(array) - _open(array)
 
         if not np.any(diff):
             return 0.0
 
         return np.linalg.norm(diff) / np.sqrt(diff.size)
 
-    return _erosion_dilation_penalty
+    @classmethod
+    def from_params(
+        cls,
+        radius: Union[float, Tuple[float, ...]] = None,
+        dl: Union[float, Tuple[float, ...]] = None,
+        *,
+        size_px: Union[int, Tuple[int, ...]] = None,
+        beta: float = 100.0,
```

Comment:
in my recent experience, this is probably too high. I might recommend like 20.. not sure if this breaks backwards compatibility technically

Reply:
It would, because that default is used in a bunch of notebooks 😀 should we change it anyways?

Reply:
probably... but perhaps a goal before 2.8 should be to figure out very good defaults for these things.